src/GoTrueApi.ts

```diff
@@ -151,7 +151,9 @@ export default class GoTrueApi {
   ): Promise<{ user: User | null; data: User | null; error: Error | null }> {
     try {
       const data: any = await get(`${this.url}/user`, { headers: this._createRequestHeaders(jwt) })
-      return { user: data, data, error: null }
+      let user = data 
+      if (data.new_email) user.email = data.new_email
+      return { user, data, error: null }
```

Comment:
This is one way we can resolve the strange behavior

we might even want to add an `initial_email` field which stores the email that the user first used during signup? I think it's even better to be added to the `app_data`


```
{
  "app_metadata": Object {
    "provider": "email",
    "initial_email": "user@email.com"
  },
  "email": "updated@email.com",         // original email
  "id": Any<String>,
  "role": "",
}
```

Reply:
this solution seems great! 

In a future version it would be cool to differentiate the email from the user login, and to be able to choose a username.